src/service/navigation.js

```diff
@@ -409,6 +422,31 @@ export class Navigation {
     }
   }
 
+  /**
+   *
+   * @param {number} priority
+   * @return {boolean}
+   */
+  isLinkRulePriorityUsed_(priority) {
+    for (let i = 0; i < this.registeredLinksRules_.length; i++) {
+      const lr = this.registeredLinksRules_[i];
+      if (lr.priority_ === priority) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @param {!Function} callback
+   * @param {number} priority
+   */
+  registerLinkRule(callback, priority) {
+    user().assert(!this.isLinkRulePriorityUsed_(priority),
+        'Rule with same priority is already in use.');
```

Comment:
I think maybe we should allow more than one rule with the same priority. If not, every time a new use case is introduced we will have to add another entry to the enum.

Reply:
right. currently..any number is open for anyone to use. I initially thought of reserving numbers for each extension or use case but as you mentioned, I see no issue with allowing some rules to have the same priority or maybe have a concept of no priority and it can be executed in any order..left for last. As discussed from the meeting we need to prioritize the linker URL transformations vs the skimlinks URL transformations. If so then perhaps, reserve a number now for the linker link rule priority? and make priority optional.

Reply:
I think we need priority uniqueness. Otherwise, isn't the order of invocation for handlers with the same priority indeterminate?

Reply:
right..I think we should have priority uniqueness but was concerned with users who want to register an anchor mutation and didn't care for priority, for example if simply adding a queryParam with no concern of where it's placed. But we can start with reserving a priority for users of this and change it if users in the future show similar concerns.

@calebcordry When using the API, just define a number and I can follow with a separate PR with the priority enum, eslint rules and/or proper js annotations to ensure that users define and reserve a priority number in an enum.

Reply:
Agree, I was thinking that people may not care specific order just that one group executes before the other. I guess we can cross that bridge when when see more than two use cases 🙂. 

@alabiaga Sounds good.